### ISSUE TITLE
[Bug]: aws_acm_certificate updating on each tf apply

### ISSUE THREAD

**Issue by JHBoricua** [hashicorp/terraform-provider-aws#38059]
### Terraform Core Version

1.8.5

### AWS Provider Version

5.54.1

### Affected Resource(s)

Terraform wants to update aws_acm_certificate resource in-place on each apply despite no changes being made on resource.

### Expected Behavior

Terraform should see resource as unchanged and take no action.

### Actual Behavior

Terraform reports it will update the resource in-place.

### Relevant Error/Panic Output Snippet
The output doesn't indicate why Terraform thinks the cert needs updating, hence the bug report.

```shell
Terraform will perform the following actions:

  # aws_acm_certificate.mycert will be updated in-place
  ~ resource "aws_acm_certificate" "mycert" {
        id                        = "arn:aws:acm:us-east-1:##########:certificate/...."
        tags                      = {
            "Name" = "my cert name tag"
        }
        # (16 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```
### Terraform Configuration Files

Resource configured as follows:
```
resource "aws_acm_certificate" "mycert" {
  provider                  = aws.WORKLOAD
  certificate_authority_arn = "arn:aws:acm-pca:${var.AWS_REGION}:${var.AWS_ACCOUNT_ID}:certificate-authority/......."
  domain_name               = var.CERT_CN[var.SDLC_KEY]
  subject_alternative_names = [var.CERT_SAN[var.SDLC_KEY]]
  lifecycle {
    create_before_destroy = true
  }
  tags = {
    Name = "${var.NAME_PREFIX[var.SDLC_KEY]}-${var.APPLICATION}-certificate"
  }
}
```
### Steps to Reproduce

Create certificate using aws_acm_certificate resource. Do terraform plan or apply after cert creation. Terraform will report it needs to update cert in-place.

### Important Factoids

This is happening when issuing both public and private certs using the aws_acm_certificate resource.

### Would you like to implement a fix?

None

**Comment by JHBoricua** [comment]
@assafdori But I don't have the root domain in the SAN parameter, only the SAN value. That said, It looks as if this might have been resolved in the 5.55.0 release.

**Comment by justinretzolk** [comment]
Since it sounds like this was resolved, I'll close this out. Thank you for taking the time to raise this! Please do let us know if you run into any unexpected behavior in the future!
